ibis/pandas/core.py

```diff
@@ -324,6 +381,18 @@ def main_execute(expr, params=None, scope=None, aggcontext=None, **kwargs):
     if scope is None:
         scope = {}
 
+    if timecontext is None:
+        timecontext = {}
+    else:
+        try:
+            timecontext = tuple(map(pd.to_datetime, timecontext))
+        except ValueError:
+            raise com.IbisError(
+                'Cannot resolve timecontext for expression\n{}.'.format(
+                    expr.get_name()
+                )
+            )
```

Comment:
do we have tests for this?

Reply:
Yes, I added a `test_bad_timecontext` test in `test_timecontext.py` for all bad inputs.